File: src/main/java/com/fluxchess/pulse/MoveType.java
/*
 * Copyright (C) 2013-2015 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */
package com.fluxchess.pulse;

final class MoveType {

  static final int MASK = 0x7;

  static final int NORMAL = 0;
  static final int PAWNDOUBLE = 1;
  static final int PAWNPROMOTION = 2;
  static final int ENPASSANT = 3;
  static final int CASTLING = 4;

  static final int NOMOVETYPE = 5;

  private MoveType() {
  }

  static boolean isValid(int type) {
    switch (type) {
      case NORMAL:
      case PAWNDOUBLE:
      case PAWNPROMOTION:
      case ENPASSANT:
      case CASTLING:
        return true;
      default:
        return false;
    }
  }

}

File: pulse-kotlin/src/commonMain/kotlin/com/fluxchess/pulse/kotlin/MoveType.kt
/*
 * Copyright 2013-2024 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */

package com.fluxchess.pulse.kotlin

typealias MoveType = Int

const val NORMAL_MOVE: MoveType = 0
const val PAWN_DOUBLE_MOVE: MoveType = 1
const val PAWN_PROMOTION_MOVE: MoveType = 2
const val EN_PASSANT_MOVE: MoveType = 3
const val CASTLING_MOVE: MoveType = 4

const val NO_MOVE_TYPE: MoveType = 5

val moveTypes = intArrayOf(NORMAL_MOVE, PAWN_DOUBLE_MOVE, PAWN_PROMOTION_MOVE, EN_PASSANT_MOVE, CASTLING_MOVE)
